src/libraries/System.Net.Mail/src/System/Net/Mail/MailWriter.cs

```diff
@@ -28,11 +28,11 @@ internal override void WriteHeaders(NameValueCollection headers, bool allowUnico
             if (headers == null)
                 throw new ArgumentNullException(nameof(headers));
 
-            foreach (string? key in headers) // TODO-NULLABLE: https://github.com/dotnet/csharplang/issues/3214
+            foreach (string key in headers)
             {
                 string[] values = headers!.GetValues(key)!;
```

Comment:
NIT: not related to this PR change, but i think we can remove `!` from `headers!`

Reply:
Addressed